### PR TITLE
Link Channel to the stored contacts

### PR DESCRIPTION
**Pull request by shubhamkmr04**
This PR addresses https://github.com/ZeusLN/zeus/issues/2082

It adds functionality to link the `remotePubkey` in the Channel view to a contact from our stored contacts.

It retrieves contacts from the encrypted storage, matches the `remotePubkey` with contact pubkey, and displays the contact name next to remotePubkey if a match is found. 

Check out the demo picture below addressing the change

Simulator Screenshot - iPhone 15 Pro - 2024-07-29 at 14 43 20

### CODE REVIEW

**Comment by kaloudis** [comment]
Can you please add a description of the change and some screenshots or video illustrating the change?

**Comment by shubhamkmr04** [comment]
> Can you please add a description of the change and some screenshots or video illustrating the change?

added a description and demo video

File: views/Channels/Channel.tsx [hunk 1]
```diff
@@ -295,6 +364,17 @@ export default class ChannelView extends React.Component<
                                 </Text>
                             </TouchableOpacity>
                         )}
+                        <Text
```
Comment (kaloudis) [inline views/Channels/Channel.tsx:367]: I really like the display component here but i don't think we need the label. Let's remove `Matching Contact Found`

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: views/Channels/Channel.tsx (hunk 1 above)
Comment (shubhamkmr04) [reply views/Channels/Channel.tsx:367]: sure

File: locales/en.json [hunk 2]
```diff
@@ -432,6 +432,7 @@
     "views.Channel.Total.offline": "Total offline",
     "views.Channel.zeroConf": "Zero conf",
     "views.Channel.commitmentType": "Commitment Type",
+    "views.Channel.matchingContactFound": "Matching Contact Found",
```
Comment (kaloudis) [inline locales/en.json:435]: we're no longer using this locale now

**Review by kaloudis — CHANGES_REQUESTED** [review]

**Comment by kaloudis** [comment]
Initial tests look good. We're gonna have to refactor this with the `ChannelsStore` changes though. We'll try to get that in soon.

File: views/Channels/Channel.tsx [hunk 3]
```diff
@@ -75,14 +79,79 @@ export default class ChannelView extends React.Component<
...
         if (BackendUtils.isLNDBased() && channel.channelId != null) {
             ChannelsStore.loadChannelInfo(channel.channelId);
         }
     }
 
+    componentDidMount() {
+        this.loadContacts();
+    }
+
+    loadContacts = async () => {
```
Comment (kaloudis) [inline views/Channels/Channel.tsx:95]: Please rebase and refactor to use the newly added `ContactStore`

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: views/Channels/Channel.tsx [hunk 4]
```diff
@@ -83,6 +87,57 @@ export default class ChannelView extends React.Component<
         }
     }
 
+    componentDidMount() {
+        const { ContactStore } = this.props;
+        const { loadContacts } = ContactStore;
+        loadContacts();
```
Comment (kaloudis) [inline views/Channels/Channel.tsx:93]: Why do we have to load contacts again on page load? Aren't they accessible in the store already?

Comment (shubhamkmr04) [reply views/Channels/Channel.tsx:93]: yeahh. you're right. I'll change that

**Review by kaloudis — APPROVED** [review]
tACK

**Pull request merged**